daemon/algod/api/server/v2/handlers.go

```diff
@@ -442,14 +442,17 @@ func (v2 *Handlers) PendingTransactionInformation(ctx echo.Context, txid string,
 
 	// Encoding wasn't working well without embedding "real" objects.
 	response := struct {
-		AssetIndex      *uint64                `codec:"asset-index,omitempty"`
-		CloseRewards    *uint64                `codec:"close-rewards,omitempty"`
-		ClosingAmount   *uint64                `codec:"closing-amount,omitempty"`
-		ConfirmedRound  *uint64                `codec:"confirmed-round,omitempty"`
-		PoolError       string                 `codec:"pool-error"`
-		ReceiverRewards *uint64                `codec:"receiver-rewards,omitempty"`
-		SenderRewards   *uint64                `codec:"sender-rewards,omitempty"`
-		Txn             transactions.SignedTxn `codec:"txn"`
+		AssetIndex       *uint64                        `codec:"asset-index,omitempty"`
+		ApplicationIndex *uint64                        `codec:"application-index,omitempty"`
+		CloseRewards     *uint64                        `codec:"close-rewards,omitempty"`
+		ClosingAmount    *uint64                        `codec:"closing-amount,omitempty"`
+		ConfirmedRound   *uint64                        `codec:"confirmed-round,omitempty"`
+		GlobalStateDelta *generated.StateDelta          `codec:"global-state-delta,omitempty"`
+		LocalStateDelta  *[]generated.AccountStateDelta `codec:"local-state-delta,omitempty"`
+		PoolError        string                         `codec:"pool-error"`
+		ReceiverRewards  *uint64                        `codec:"receiver-rewards,omitempty"`
+		SenderRewards    *uint64                        `codec:"sender-rewards,omitempty"`
+		Txn              transactions.SignedTxn         `codec:"txn"`
```

Comment:
This is definitely something I want to revisit. There is a generated type for this object, but instead of `transactions.SignedTxn` it uses the generic `map[string]interface{}`. So far I have been unable to properly convert a SignedTxn with the generic type. I expect that it is possible, and when initially implementing this I got it working for single-level objects like BlockHeader.

For now, I prefer to leave the ugly anonymous struct as a reminder that this is a hack and should be fixed properly.